src/jou_compiler.h

```diff
@@ -324,6 +327,7 @@ struct Type {
         TYPE_UNSIGNED_INTEGER,
         TYPE_BOOL,
         TYPE_DOUBLE,
+        TYPE_FLOAT,
```

Comment:
I think we don't need both `TYPE_DOUBLE` and `TYPE_FLOAT` here, similarly to how `int` and `long` both use `TYPE_SIGNED_INTEGER`. IMO it would be cleaner to distinguish them using `width_in_bits`, like we already do for integers.

Reply:
Actually, I'm no longer sure about this. We will only ever have two float-like types anyway, but we have many different kinds of integers.

It's fine in both ways :)

Reply:
So, we just need TYPE_DOUBLE?

Reply:
That would be one way. But as I said, it's also fine to keep it like you have now and have `TYPE_FLOAT` and `TYPE_DOUBLE`.

Reply:
ok